File: Stack/valid_parentheses.py

def valid_parentheses(s: str) -> bool:
    pairs = {
        ")": "(",
        "}": "{",
        "]": "["
    }

    stack = []

    for char in s:
        if char not in pairs:
            stack.append(char)
            continue
        if not stack or stack[-1] != pairs[char]:
            return False
        stack.pop()
    return not stack

File: Stack/validParentheses.js
var isValid = function (s) {
  let stack = [];
  const OPENING_BRACES = "([{";
  const CLOSING_BRACES = ")]}";
  const PAIRS = {
    "{": "}",
    "[": "]",
    "(": ")",
  };

  for (let i = 0; i < s.length; i++) {
    if (OPENING_BRACES.includes(s[i])) {
      stack.push(s[i]);
    } else if (CLOSING_BRACES.includes(s[i])) {
      if (!stack.length) {
        return false;
      }
      if (PAIRS[stack.pop()] !== s[i]) {
        return false;
      }
    }
  }

  return !stack.length;
};

console.log(isValid("()[]{}"));
